soyjs/exec.go

```diff
@@ -510,44 +511,119 @@ func (s *state) visitSwitch(node *ast.SwitchNode) {
 	s.jsln("}")
 }
 
-// TODO: {msg} node translation is unimplemented
 func (s *state) visitMsg(node *ast.MsgNode) {
-	var pluralNode, ok = node.Body.Children()[0].(*ast.MsgPluralNode)
-	if !ok {
-		s.visitMsgBody(node)
+	// If no bundle was provided, walk the message sub-nodes.
+	if s.options.Messages == nil {
+		s.visitMsgNode(node)
 		return
 	}
 
-	s.jsln("switch (", pluralNode.Value, ") {")
+	// Look up the message in the bundle.
+	var msg = s.options.Messages.Message(node.ID)
+	if msg == nil {
+		s.visitMsgNode(node)
+		return
+	}
+
+	// Render each part.
+	s.evalMsgParts(node, msg.Parts)
+}
+
+func (s *state) evalMsgParts(msgNode *ast.MsgNode, parts []soymsg.Part) {
+	for _, part := range parts {
+		switch part := part.(type) {
+
+		case soymsg.RawTextPart:
+			s.writeRawText([]byte(part.Text))
+
+		case soymsg.PlaceholderPart:
+			// Find the node corresponding to the placeholder, and walk it.
+			var phnode = msgNode.Placeholder(part.Name)
+			if phnode == nil {
+				s.errorf("failed to find placeholder %q in %q",
+					part.Name, soymsg.PlaceholderString(msgNode))
+			}
+			s.walk(phnode.Body)
+
+		case soymsg.PluralPart:
+			// Find the corresponding node for this part.
+			child := s.findPluralNode(msgNode, part.VarName)
+			s.jsln("switch (", child.Value, ") {")
```

Comment:
(as mentioned in the other comment, the value in the switch should be selectorFn(n), where "n" is the number of items and selectorFn is the thing that returns the plural case for the number of items)